src/components/pages/DataViz/RenderMap.js

```diff
@@ -49,23 +55,89 @@ const RenderMap = () => {
     setDetailsData(null);
   });
 
-  let geojson = {
+  let geojsonComplete = {
+    type: 'FeatureCollection',
+    features: [],
+  };
+  let geojsonRejected = {
+    type: 'FeatureCollection',
+    features: [],
+  };
+  let geojsonConfirmed = {
+    type: 'FeatureCollection',
+    features: [],
+  };
+  let geojsonIdentified = {
+    type: 'FeatureCollection',
+    features: [],
+  };
+  let geojsonProspecting = {
+    type: 'FeatureCollection',
+    features: [],
+  };
+  let geojsonUnderConstruction = {
     type: 'FeatureCollection',
     features: [],
   };
-  let featureCollection = [];
 
   //this will run function after brindges will be filtered
   function certainBridgeShows(bridges) {
-    bridges.forEach(bridge =>
-      featureCollection.push({
-        type: 'Feature',
-        geometry: {
-          type: 'Point',
-          coordinates: [bridge.long, bridge.lat],
-        },
-      })
-    );
+    bridges.forEach(bridge => {
```

Comment:
It may be more efficient and easier to read having the if statements here being replaced with a branch table implementation.